datafusion/core/src/datasource/view.rs

```diff
@@ -342,6 +355,65 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn query_view_with_filter_and_limit() -> Result<()> {
+        let session_ctx = SessionContext::with_config(
+            SessionConfig::new().with_information_schema(true),
+        );
+
+        session_ctx
+            .sql("CREATE TABLE abc AS VALUES (1,2,3), (4,5,6), (7,8,9)")
+            .await?
+            .collect()
+            .await?;
+
+        let view_sql = "CREATE VIEW xyz AS SELECT column1, column2 FROM abc";
+        session_ctx.sql(view_sql).await?.collect().await?;
+
+        let results = session_ctx.sql("SELECT * FROM information_schema.tables WHERE table_type='VIEW' AND table_name = 'xyz'").await?.collect().await?;
+        assert_eq!(results[0].num_rows(), 1);
+
+        let result_plan = session_ctx
+            .sql("EXPLAIN CREATE VIEW efg as SELECT column1 FROM xyz WHERE column2 % 2 = 0 OFFSET 1 LIMIT 2")
```

Comment:
I think what we want here is to just select from the view. 

```suggestion
            .sql("EXPLAIN SELECT column1 FROM xyz WHERE column2 % 2 = 0 OFFSET 1 LIMIT 2")
```

Reply:
When we select from a view, it will cause the `TableView::scan` method to be called.

Reply:
Thank you, Andy! I will update the test.

Reply:
I got this:
```
[
    "+---------------+-----------------------------------------------------------------------------------+",
    "| plan_type     | plan                                                                              |",
    "+---------------+-----------------------------------------------------------------------------------+",
    "| logical_plan  | Limit: skip=1, fetch=2                                                            |",
    "|               |   Projection: #xyz.column1                                                        |",
    "|               |     Filter: #xyz.column2 % Int64(2) = Int64(0)                                    |",
    "|               |       TableScan: xyz projection=[column1, column2]                                |",
    "| physical_plan | GlobalLimitExec: skip=1, fetch=2                                                  |",
    "|               |   CoalescePartitionsExec                                                          |",
    "|               |     ProjectionExec: expr=[column1@0 as column1]                                   |",
    "|               |       CoalesceBatchesExec: target_batch_size=4096                                 |",
    "|               |         FilterExec: column2@1 % 2 = 0                                             |",
    "|               |           RepartitionExec: partitioning=RoundRobinBatch(10)                       |",
    "|               |             GlobalLimitExec: skip=None, fetch=None                                |",
    "|               |               CoalescePartitionsExec                                              |",
    "|               |                 ProjectionExec: expr=[column1@0 as column1, column2@1 as column2] |",
    "|               |                   RepartitionExec: partitioning=RoundRobinBatch(10)               |",
    "|               |                     MemoryExec: partitions=1, partition_sizes=[1]                 |",
    "|               |                                                                                   |",
    "+---------------+-----------------------------------------------------------------------------------+",
]
```
Seems like only `projection` is pushed down, but not `filter` or `limit`. Not sure what goes wrong 😅

Reply:
Ah, the test is using a `MemTable`, which does not support filter pushdown. We can see that the filter and limit are added into the logical plan though so I think this is good enough for this PR.

Reply:
> Ah, the test is using a MemTable 

Thank you for the explanation Andy. Is there a way to test using other data source except the `MemTable`?

Reply:
> We can see that the filter and limit are added into the logical plan though so I think this is good enough for this PR.

I guess what we do in this PR is to add the `limit` and `filter` information into the `TableScan`. However, I can't find a way to test whether it works properly. Could you help me?